src/Tools/dotnet-trace/dotnet-trace.csproj

```diff
@@ -2,28 +2,42 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-
-    <!-- Target .NET Core 2.1 so it will run on LTS -->
     <TargetFramework>netcoreapp2.1</TargetFramework>
+    <RootNamespace>Microsoft.Diagnostics.Tools.Trace</RootNamespace>
+    <IsPackable>False</IsPackable>
+    <PackAsTool>False</PackAsTool>
+  </PropertyGroup>
 
-    <RootNamespace>Microsoft.Diagnostics.Tools.Collect</RootNamespace>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-
-    <!-- Don't pack until ship engineering is done. Currently causing the official job to fail.
-    <IsPackable>true</IsPackable>
-    <PackAsTool>true</PackAsTool>
-    -->
+  <PropertyGroup>
+    <UseSharedCompilation>False</UseSharedCompilation>
+    <LangVersion>Latest</LangVersion>
+    <WarningLevel>4</WarningLevel>
+    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+    <GenerateDocumentationFile>False</GenerateDocumentationFile>
   </PropertyGroup>
 
   <ItemGroup>
-    <Compile Include="..\Common\CommandLineException.cs" Link="CommandLineException.cs" />
-    <Compile Include="..\Common\ConsoleCancellation.cs" Link="ConsoleCancellation.cs" />
-    <Compile Include="..\Common\DebugUtil.cs" Link="DebugUtil.cs" />
+    <PackageReference Include="System.CommandLine.Experimental" Version="0.2.0-alpha.19167.2" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Remove="ConfigPathDetector.cs" />
+    <Compile Remove="EtwCollector.cs" />
+    <Compile Remove="EventCollector.cs" />
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="McMaster.Extensions.CommandLineUtils" Version="2.2.5" />
-    <PackageReference Include="Microsoft.Diagnostics.Tracing.TraceEvent" Version="2.0.30" />
+    <ProjectReference Include="$(MSBuildThisFileDirectory)\..\..\Microsoft.Diagnostics.Tracing.Eventing\EventPipeClient.csproj" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <!--
+      Suppress warnings for using preview SDK:
+      You are working with a preview version of the .NET Core SDK.
+      You can define the SDK version via a global.json file in the current project.
+      More at https://go.microsoft.com/fwlink/?linkid=869452
+    -->
+    <SuppressNETCoreSdkPreviewMessage>True</SuppressNETCoreSdkPreviewMessage>
```

Comment:
Not sure why you need this.

Reply:
When I build from the folder, with the latest .NET Core that is on my PATH then I get warnings. This is to suppress it.

Reply:
I guess I always use the "build.cmd/build.sh" script (the arcade way) in the root which should use the SDK installed in .dotnet.  

So you are doing "dotnet build dotnet-trace.csproj" in the project directory which is ok but you could be building with a different SDK than the CI and official builds.